integration-tests/cloud-integration-tests/bigquery/bigquery_test.go

```diff
@@ -70,195 +72,423 @@ func TestBigQueryWorkflows(t *testing.T) {
+							},
+							Asserts: []func(*e2e.Task) error{
+								e2e.AssertByExitCode,
+							},
+						},
+						{
+							Name:    "scd2-by-time: run products_updated_02.sql with SCD2 materialization",
+							Command: binary,
+							Args:    append(append([]string{"run"}, configFlags...), "--env", "default", filepath.Join(tempDir, "test-pipelines/scd2-pipelines/scd2-by-time-pipeline/assets/products.sql")),
+							Env:     []string{},
+							Expected: e2e.Output{
+								ExitCode: 0,
+							},
+							Asserts: []func(*e2e.Task) error{
+								e2e.AssertByExitCode,
+							},
+						},
+						{
+							Name:    "scd2-by-time: query the updated table 02",
+							Command: binary,
+							Args:    append(append([]string{"query"}, configFlags...), "--env", "default", "--asset", filepath.Join(tempDir, "test-pipelines/scd2-pipelines/scd2-by-time-pipeline/assets/products.sql"), "--query", "SELECT product_id,product_name,stock,_is_current,_valid_from FROM test.products ORDER BY product_id, _valid_from;", "--output", "csv"),
+							Env:     []string{},
+							Expected: e2e.Output{
+								ExitCode: 0,
+								CSVFile:  filepath.Join(tempDir, "test-pipelines/scd2-pipelines/scd2-by-time-pipeline/expectations/scd2_by_time_expected_update_02.csv"),
+							},
+							Asserts: []func(*e2e.Task) error{
+								e2e.AssertByExitCode,
+								e2e.AssertByCSV,
+							},
+						},
+						{
+							Name:    "scd2-by-time: drop the table (expect error but table will be dropped)",
+							Command: binary,
+							Args:    append(append([]string{"query"}, configFlags...), "--connection", "athena-default", "--query", "DROP TABLE IF EXISTS test.products;"),
+							Env:     []string{},
+							Expected: e2e.Output{
+								ExitCode: 0,
+							},
+							Asserts: []func(*e2e.Task) error{
+								e2e.AssertByExitCode,
+							},
 						},
-						Asserts: []func(*e2e.Task) error{
-							e2e.AssertByExitCode,
+						{
+							Name:    "scd2-by-time: confirm the table is dropped",
+							Command: binary,
+							Args:    append(append([]string{"query"}, configFlags...), "--connection", "athena-default", "--query", "SELECT * FROM test.products;"),
+							Env:     []string{},
+							Expected: e2e.Output{
+								ExitCode: 1, // Should fail because table doesn't exist
+							},
+							Asserts: []func(*e2e.Task) error{
+								e2e.AssertByExitCode,
+							},
 						},
 					},
-				},
+				}
 			},
 		},
 	}
+
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			t.Parallel()
 
-			err := tt.workflow.Run()
-			require.NoError(t, err, "Workflow %s failed: %v", tt.workflow.Name, err)
+			tempDir, tempErr := os.MkdirTemp("", "bigquery-test-*")
+			require.NoError(t, tempErr, "Failed to create temporary directory")
+			defer os.RemoveAll(tempDir)
+
+			srcPipelines := filepath.Join(currentFolder, "test-pipelines")
+			destPipelines := filepath.Join(tempDir, "test-pipelines")
+			copyErr := copyDir(srcPipelines, destPipelines)
```

Comment:
you can reuse the `err` variable. There's no need to create separate error variables, unless you need to reference errors later.

```go
tmpDir, err := os.MkdirTemp("", "bigquery-test-*")
if err != nil { ... }

err = copyDir(srcPipelines, destPipelines)
if err != nil { ... }
```

is perfectly idiomatic code.